template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateValidatorTest.java

```diff
@@ -353,6 +360,44 @@ public void testValidationIfNodeManagerCountWillBeHigherThanZeroInTheSameGroup()
                 Optional.of(clouderaManagerRepo), hostGroup, -2, Set.of(compute, worker));
     }
 
+    @Test
+    public void checkKafkaBrokerBlackListedUpscaleShouldBeAllowed() {
+        String hostGroup = "compute";
+        ClouderaManagerProduct clouderaManagerRepo = new ClouderaManagerProduct();
+        clouderaManagerRepo.setVersion("7.2.12");
+        CmTemplateProcessor processor = mock(CmTemplateProcessor.class);
+        Map<String, Set<String>> componentsByHostGroup = new HashMap<>();
+        Set<String> components = new HashSet<>();
+        components.add("KAFKA_BROKER");
+        componentsByHostGroup.put("compute", components);
+        when(processor.getComponentsByHostGroup()).thenReturn(componentsByHostGroup);
+        Optional<ClouderaManagerProduct> product = Optional.of(clouderaManagerRepo);
+        subject.validateBlackListedScalingRoles(ACCOUNT_ID, processor, hostGroup, 1, product);
+    }
+
+    @Test
+    public void checkKafkaBrokerBlackListedUpscaleShouldBeDenied() {
+        String hostGroup = "compute";
+        ClouderaManagerProduct clouderaManagerRepo = new ClouderaManagerProduct();
+        clouderaManagerRepo.setVersion("7.2.6");
+        CmTemplateProcessor processor = mock(CmTemplateProcessor.class);
+        Map<String, Set<String>> componentsByHostGroup = new HashMap<>();
+        Set<String> components = new HashSet<>();
+        components.add("KAFKA_BROKER");
+        componentsByHostGroup.put("compute", components);
+        when(processor.getComponentsByHostGroup()).thenReturn(componentsByHostGroup);
+        Optional<ClouderaManagerProduct> product = Optional.of(clouderaManagerRepo);
+        Throwable exception = Assert.assertThrows(BadRequestException.class,
+                () -> subject.validateBlackListedScalingRoles(ACCOUNT_ID, processor, hostGroup, 1, product));
+        assertEquals("'KAFKA_BROKER' service is not enabled to scale until CDP 7.2.12", exception.getMessage());
+    }
+
+    @Test
+    public void isVersionEnablesScaling() {
```

Comment:
I would simplify it with a parameterized test, also if we have a cutover on the version for a feature, I would test 1 or 2 versions less, the cutover version, and 1 or 2 versions more.